skymp5-server/cpp/localization_provider/LocalizationProvider.cpp

```diff
@@ -0,0 +1,133 @@
+#include "LocalizationProvider.h"
+#include <fstream>
+
+DirectoryEntry::DirectoryEntry(uint32_t stringId, uint32_t stringOffset)
+  : stringId(stringId)
+  , offset(stringOffset)
+{
+}
+
+std::vector<DirectoryEntry*> LocalizationProvider::ParseDirectoryEntries(
+  std::vector<char> buffer)
+{
+  std::vector<DirectoryEntry*> directoryEntries;
+  uint32_t numberOfEntries = *(uint32_t*)&buffer[0];
+
+  directoryEntries.resize(numberOfEntries);
+
+  uint32_t offset = 8;
+
+  for (int i = 0; i < numberOfEntries; i++) {
+    uint32_t stringId = *(uint32_t*)&(buffer[offset]);
+    offset += 4;
+
+    uint32_t stringOffset = *(uint32_t*)&(buffer[offset]);
+    offset += 4;
+
+    DirectoryEntry* entry = new DirectoryEntry(stringId, stringOffset);
+    directoryEntries[i] = entry;
+  }
+
+  return directoryEntries;
+}
+
+std::map<uint32_t, std::string> LocalizationProvider::ParseStrings(
+  std::vector<char> buffer, std::vector<DirectoryEntry*> entries)
+{
+  std::map<uint32_t, std::string> localization;
+  uint32_t numberOfEntries = entries.size();
+
+  for (int i = 0; i < numberOfEntries; i++) {
+    uint32_t start = 8 + numberOfEntries * 8 + (*entries[i]).offset;
+
+    for (int charIndex = 0; charIndex < buffer.size() - start; charIndex++) {
+      if (buffer[start + charIndex] == 0) {
+        break;
+      }
+
+      (*entries[i]).str += buffer[start + charIndex];
+    }
+
+    localization[(*entries[i]).stringId] = (*entries[i]).str;
+  }
+
+  return localization;
+}
+
+std::map<uint32_t, std::string> LocalizationProvider::ParseILDLStrings(
+  std::vector<char> buffer, std::vector<DirectoryEntry*> entries)
+{
+  std::map<uint32_t, std::string> localization;
```

Comment:
Bad thing to use same name for class field and variable in class function. Also you can fill class field directly from this function.